src/darkleaf/di/core.clj

```diff
@@ -797,3 +797,19 @@
                ~@body))
            (finally
              (.close resource#)))))))
+
+(defn log [built-cb demolished-cb]
```

Comment:
"A logging middleware. Calls :after-build! and :after-demolish! during di/start. Must be the last one in the middleware chain.
Both callbacks are expected to accept the following arg {:keys [key obj]}."